### ISSUE TITLE
Just saying that "this work is not copyrighted" is not going to cut it in terms of placing this code into an effective worldwide public domain.

### ISSUE THREAD

**Issue by hobbledehoy899** [jasonrohrer/OneLife#6]
[Here's a really good write-up as to why that is.](https://www.gnu.org/licenses/license-list.en.html#NoLicense)

**Comment by jasonrohrer** [comment]
You mean because I don't say "public domain" explicitly?  I don't believe in copyright, so I can't use copyleft, because that depends on copyright and seeks to control the actions of other people.

I'm not a huge fan of CC0 because I don't think it's necessary, and Creative Commons itself will not be around forever.  Posting on that particular website means nothing to me.

I'd rather have a notice with the work itself, which I have.

Do you have a suggestion for how to improve the wording of my notice?  I don't want it to be too long.  The public domain is simple, so I want my wording to be simple.

**Comment by hobbledehoy899** [comment]
Either say "public domain" explicitly (in which case the work will still not be fully released the way you want it to be because not all legal systems have processes for reliably donating works to the public domain) or just go with CC0 or an equally permissive license.

**Comment by romlok** [comment]
I agree with the issue reporter that the statement you give is almost certainly not enough for it to have the legally-certain effect it seems you want it to have.

> I'm not a huge fan of CC0 because I don't think it's necessary, and Creative Commons itself will not be around forever. Posting on that particular website means nothing to me.

Creative Commons isn't an image hosting site or content directory (though they do have a search tool for some such sites), it is an organisation which creates and promotes use of the Creating Commons licenses. All one needs from them is to choose a license and apply it to your creative work. The organisation may indeed go away at some point, but the licenses won't cease to apply.

> Do you have a suggestion for how to improve the wording of my notice? I don't want it to be too long. The public domain is simple, so I want my wording to be simple.

Unfortunately, "public domain" is not actually all that simple from an international legal standpoint. If it were that simple, there would be no need for CC0 - which was crafted specifically for artists who want to do what you want to do.

As [Creative Commons explain](https://creativecommons.org/share-your-work/public-domain/cc0):

> The Problem
> Dedicating works to the public domain is difficult if not impossible for those wanting to contribute their works for public use before applicable copyright or database protection terms expire. Few if any jurisdictions have a process for doing so easily and reliably. Laws vary from jurisdiction to jurisdiction as to what rights are automatically granted and how and when they expire or may be voluntarily relinquished. More challenging yet, many legal systems effectively prohibit any attempt by these owners to surrender rights automatically conferred by law, particularly moral rights, even when the author wishing to do so is well informed and resolute about doing so and contributing their work to the public domain.


So my suggestion would be to create a README file, in which you include a standard copyright notice (you have the copyright, even if you don't want it), and state that you are licensing everything under Creative Commons Zero (CC0). Then link to https://creativecommons.org/publicdomain/zero/1.0/ and/or include [the text of the license](https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt) as a LICENSE file in the repo.

Disclaimer: I'm not a copyright lawyer, but I play one on the Internet.

EDIT: Creative Commons have a [template copyright notice you can use](https://wiki.creativecommons.org/wiki/CC0_FAQ#May_I_apply_CC0_to_computer_software.3F_If_so.2C_is_there_a_recommended_implementation.3F)

**Comment by jasonrohrer** [comment]
Well, I'm also just not that into lots of legalese.

I wouldn't just link to CC0, because like I said, the Creative Commons won't be around forever.  The link will break someday.

So I'd need to include the license, which is a load of text that I don't even want to read myself.

Copyright, as far as I'm aware, is only enforced by the "offended party" choosing to enforce it in civil court.  I.e., it's a matter of civil, not criminal, law.

I know that I have no "right" to wave my copyright according to current law.

But I just don't believe in copyright.  Copyright is nonsense.

So I don't want to get involved with it, or leverage "licenses" of any kind, because those all depend on copyright for their power.  I can't grant you a license if I have no power in the first place.  I don't believe I have the power to grant you anything.  I believe you are a free person, and that I have no control over your private or public actions outside of my immediate physical sphere.

So, you'll really just have to take my word for it.  I'm not going to leverage copyright to come after you.

And just because I said "CC0" wouldn't change that.  When has CC0 stood up in court when a creator tried to revoke it later?

I think a creator probably has the "power" to revoke a license under current law.  If not, show me the court precedent.  And even if you show me a precedent, show me one in every country.

I just see the whole thing as pretty meaningless.

I'm not going to come after you.  You'll just have to take my word for it.

**Comment by jasonrohrer** [comment]
And I don't mind using the term "public domain" in my text file, if that will help.  But I don't think it will help.

**Comment by Shaac** [comment]
Another option would be to use http://unlicense.org/ by adding a short file `COPYING`, `LICENSE`, or `UNLICENSE` in your root folder.

That way you don’t have to register on any website, you don’t have to add license on top of every source code, it seems to me from what I gathered from your viewpoint that it fits your values.

By doing so, it would be easier for people to understand that there is no copyright, even github would automatically detect it. The Free Software Fondation recognize this license, so people could use part of you code in other free software without fear.

I’m always appalled at how difficult it is to publish stuff in the public domain. The [story of SQLite](https://www.sqlite.org/copyright.html) with papers in a firesafe shows how complicated it is.

**Comment by romlok** [comment]
Yeah, the Unlicense is also a possibility, similar in intent to CC0, but possibly less legally inclusive.

> So I'd need to include the license, which is a load of text that I don't even want to read myself.

To be fair, most developers don't read the licenses they apply to their code, and generally don't need to either IMO. The advantage of using an existing license, over making up something on the spot, is that the implications are generally already understood collectively.

Fortunately, each Creative Commons license has a short, human-readable summary (eg: [CC0](https://creativecommons.org/publicdomain/zero/1.0/)) which is representative of the legal text, if you just want to get to know the intent and scope of the license.

> But I just don't believe in copyright. Copyright is nonsense.
> ...
> So, you'll really just have to take my word for it. I'm not going to leverage copyright to come after you.

Lots of man-made things are a nonsense to most, but they affect everyone whether you believe in them or not. This particular brand of nonsense (currently) lasts for life of the author + 70 years, so we'd not only have to take your word for it, but also the word of the heir/s of your estate. Which any of you could change your minds about and potentially revoke at any time.

But I think the point is that we don't *have to* just take your word for it - you could make it explicit for everyone, worldwide, by specifying an appropriate license.

> I think a creator probably has the "power" to revoke a license under current law. If not, show me the court precedent. And even if you show me a precedent, show me one in every country.

I believe revocation very much depends on the license chosen. In the case of CC0, [the Creative Commons FAQ states](https://creativecommons.org/faq/#what-happens-if-the-author-decides-to-revoke-the-cc-license-to-material-i-am-using) (emphasis mine):

> **The CC licenses are irrevocable.** This means that once you receive material under a CC license, you will always have the right to use it under those license terms, even if the licensor changes his or her mind and stops distributing under the CC license terms. Of course, you may choose to respect the licensor’s wishes and stop using the work.

Off the top of my head I don't know of any court cases involving revocation, even after [Flickr surprised a lot of artists](https://www.techdirt.com/articles/20141201/12352729288/flickr-plans-to-sell-creative-commons-photos-thats-okay.shtml) by selling prints of their CC-licensed art. But even if the authors of the licenses are wrong in their interpretation, would not a CC license be, at worst, still just as revocable as "taking your word for it"?

**Comment by Zegnat** [comment]
For what it is worth, I am a fan of [0BSD](https://spdx.org/licenses/0BSD.html) which is nice and short. Some information about its inception are included [on the Toybox licence page](http://landley.net/toybox/license.html#why). (And [a big discussion of the license](https://github.com/github/choosealicense.com/issues/464) took place on the [Choose a License](https://choosealicense.com/) repo, if you are interested in history.)

> I’m always appalled at how difficult it is to publish stuff in the public domain. The story of SQLite with papers in a firesafe shows how complicated it is.

IMHO, the SQLite story shows just how *easy* it is. Really the whole SQLite licence page drills down to this one sentence licence/dedication:

> Anyone is free to copy, modify, publish, use, compile, sell, or distribute the original SQLite code, either in source code form or as a compiled binary, for any purpose, commercial or non-commercial, and by any means.

(Which is very much like what is included in 0BSD.)

The mentioned paperwork isn’t really what puts it in the public domain, and isn’t limited to the public domain either. E.g. when [VLC changed licence](https://www.roguewave.com/resources/enterprise-blog/archive/vlc-license-change-a-lesson-in-perseverance) it also needed every developer to consent to this, having to create such a paper trail. [Contributor agreements](http://wiki.civiccommons.org/Contributor_Agreements/) are another good example of documentation like it. That’s what the SQLite paperwork comes down too as well: everyone who has a claim to the code says they abstain from exercising any of their rights to said code.

As long as @jasonrohrer is the sole contributor to One Hour One Life, he could just slap the SQLite sentence on here as a licence.

Although corporate lawyers are probably allergic to texts that aren’t well vetted, or even proven by courts, Jason’s statement above could already work:

> I’m not going to leverage copyright to come after you.

(Of course, I am not a lawyer, discretion is advised.)

**Comment by JuicyJuuce** [comment]
@jasonrohrer 

Your intentions are clearly noble. However, currently it would be foolish for another coder to use your code to build a game that they intend to sell. Here's why.

Let's say I have a great idea for an indie game and I see that your code would make a great starting point to build off of. I spend the next three years of my life building and promoting this game. Let's say its a big hit and I make ten million dollars off of it.

Now let's say that you and your immediate family die in car accident and your cousin (or whoever) inherits the rights to your code. They see my ten million dollars, have a quick chat with a lawyer, and successfully sue me for all of it because you didn't use a CC0 license.

Now it seems you are a great guy and I hope you and your family have long and happy lives. But I sure as hell am not going to bet my financial future on that.

tldr: Any coder who thinks this through is going to take a pass on using your code.

**Comment by SirBloodBuske** [comment]
can we just close this, it's not an issue.

**Comment by jasonrohrer** [comment]
Even if I do use CC0, my cousin can still sue you after I die.  My cousin could sue you BEFORE I die, or sue ME for squandering his inheritance into the public domain ahead of time. 

Where's the court precedent for CC0?

Where's the court precedent against a dude saying, "This work is not copyrighted, do whatever you want with it, no restrictions or permission necessary"?  When did that fail to hold up in court?

If there are tens of millions of dollars on the line, it WILL be tested in court.  I think that my statement is perfectly, legally clear.

I'm not a lawyer, but I have won a court case against a real lawyer, so I know how the process works.

Even if precedents exist, they just inform the judge's review of any new case that comes up.  They don't prevent legal action, nor guarantee that legal action will fail, given the particulars of a case.


The reality here is that CC0 or any other super-permissive license is rarely used for anything that has loads of commercial potential, like this video game.

It's kindof unknown territory.

**Comment by playdeezgames** [comment]
You'll be at GDC.

Recommendation:
Go see this guy(or really any of the gamedev law folks, they are there): 
http://schedule.gdconf.com/session/practical-ip-law-for-indie-developers-301-plain-scary-edition/853707

...and do what he say.

**Comment by romlok** [comment]
> It's kindof unknown territory.

The difference is that the CC0 license has been authored by a group of people who study and practise copyright law, and who have a history of writing similar licenses which have indeed been successfully upheld in court judgements, in various jurisdictions around the world.

If you're venturing into unknown territory, is it not a good idea to go with a guide who has experience with the kinds of challenges you'll be facing?

You may feel that your current statement is legally clear (despite its first sentence being factually incorrect, thanks to the Berne Convention), but if that was all that is necessary then the CC0 would be similarly concise. That the CC0 is significantly more verbose should strongly suggest that there are considerations which you weren't aware of.

**Comment by jasonrohrer** [comment]
New license reads:


This work is not copyrighted.  I place it into the public domain.

Do whatever you want with it, absolutely no restrictions, and no permission 
necessary.


Jason Rohrer
Davis, California
March 2018
